### PR TITLE
Activity histo view

### LINKED ISSUE #468: Replace the filterform with search query
The goal is to unify the web interface using the new search query. The filterform based query needs to be rewritten using the new API.

- [x] Migrate the Activity view.
- [x] Migrate the People view.
- [x] Migrate the Project view.
- [x] Migrate the Changes view.
- [x] Investigate if Hot/Cold change is worthty to be migrated. https://github.com/change-metrics/monocle/issues/527
- [x] Remove the legacy react-router and the 'BaseQueryComponent'.

### PR DESCRIPTION
**Pull request by TristanCacqueray**
Fixes #468

### CODE REVIEW

**Comment by TristanCacqueray** [comment]
- [ ] There is an issue with the changes_with_test metrics, it seems much lower than the legacy one.
- [ ] And the updates of change is higher, though that may be correct.

**Comment by TristanCacqueray** [comment]
After further investigation, the previous metric were calculated using the on_created_at range. With f86c3d8 the result is mostly identical, but it seems like we need a better system to setup those flavor.

@morucci can you please take a look, I think we can merge this PR now.

File: haskell/src/Monocle/Backend/Queries.hs [hunk 1]
```diff
@@ -594,8 +654,11 @@ getReviewStats = do
   let reviewStatsCommentCount = Just commentCount
       reviewStatsReviewCount = Just reviewCount
 
-  let reviewStatsCommentDelay = 0
-      reviewStatsReviewDelay = 0
+  firstComments <- withFilter [documentType ElkChangeCommentedEvent] firstEventOnChanges
+  firstReviews <- withFilter [documentType ElkChangeReviewedEvent] firstEventOnChanges
```
Comment (morucci) [inline haskell/src/Monocle/Backend/Queries.hs:727]: Nice !

File: haskell/src/Monocle/Backend/Queries.hs [hunk 2]
```diff
@@ -669,10 +686,40 @@ getReviewStats = do
...
         <*> fmap countToWord (countDocs qf)
 
-    getHisto' :: ELKDocType -> QueryM (V.Vector SearchPB.Histo)
-    getHisto' docType = fmap toPBHisto <$> withFilter [mkTerm "type" (toText $ docTypeToText docType)] (getHisto qf)
-    toPBHisto :: HistoEventBucket -> SearchPB.Histo
-    toPBHisto HistoEventBucket {..} =
-      let histoDate = heDate
-          histoCount = heCount
-       in SearchPB.Histo {..}
+    getHisto' docType = withDocType docType $ getHistoPB qf
```
Comment (morucci) [inline haskell/src/Monocle/Backend/Queries.hs:741]: Awesome !

File: haskell/src/Monocle/Backend/Queries.hs [hunk 3]
```diff
@@ -669,10 +686,40 @@ getReviewStats = do
...
+
+  lifecycleStatsUpdatesOfChanges <- pure 0
+  lifecycleStatsChangesWithTests <- pure 0
+  lifecycleStatsIterationsPerChange <- pure 0
+  lifecycleStatsCommitsPerChange <- pure 0
+
+  pure $ SearchPB.LifecycleStats {..}
+  where
+    qf = QueryFlavor Monocle.Search.Query.Author CreatedAt
+    getHisto' docType = withDocType docType $ getHistoPB qf
```
Comment (morucci) [inline haskell/src/Monocle/Backend/Queries.hs:798]: Could even be extracted of the where clause.

**Review by morucci — APPROVED** [review]
Here is two compares between master and your PR for the same dataset:

https://pasteboard.co/Kd5IdiL.png
https://pasteboard.co/Kd5IoXO.png

The metrics looks almost identical. For the change update yes, the legacy metric seems to be different and you seems to have found the reason.

- Granularity for the mean time looks different for the Reviews stats component.
- Self merged changes as well but 16 vs 17.

I don't think we should dig to much into those slightly different metric values.

I suggest to merge. And well impressive work !!!

**Pull request merged**